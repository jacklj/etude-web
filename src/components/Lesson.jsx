/* eslint-disable camelcase */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import moment from 'moment';
import StarRatingComponent from 'react-star-rating-component';
import DateTime from 'react-datetime';
import Select from 'react-select';

import {
  editLesson, getLesson, getLocations, getTeachers,
} from '../services/api';

const Label = styled.label`
  display: block;
`;

class Lesson extends Component {
  constructor(props) {
    super(props);
    this.state = {
      allLocations: [],
      selectedLocation: undefined,
      allTeachers: [],
      selectedTeacher: undefined,
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.onStarClick = this.onStarClick.bind(this);
    this.handleStartChange = this.handleStartChange.bind(this);
    this.handleEndChange = this.handleEndChange.bind(this);
    this.handleLocationChange = this.handleLocationChange.bind(this);
    this.handleTeacherChange = this.handleTeacherChange.bind(this);
    this.transformServerDataIntoState = this.transformServerDataIntoState.bind(this);
  }

  componentDidMount() {
    const event_id = this.props.match.params.id;
    this.setState({ event_id });

    getLocations()
      .then(allLocations => allLocations.map(location => ({
        value: location.id,
        label: location.name,
      })))
      .then(allLocations => this.setState({ allLocations }));

    getTeachers()
      .then(allTeachers => allTeachers.map(teacher => ({
        value: teacher.id,
        label: `${teacher.first_name} ${teacher.surname}`,
      })))
      .then(allTeachers => this.setState({ allTeachers }));

    getLesson(event_id).then(this.transformServerDataIntoState);
  }

  onStarClick(nextValue) {
    this.setState({ rating: nextValue });
  }

  handleChange(event) {
    const { target } = event;
    const { name, value } = target;

    this.setState({
      [name]: value,
    });
  }

  handleLocationChange(selectedLocation) {
    this.setState({ selectedLocation });
  }

  handleTeacherChange(selectedTeacher) {
    this.setState({ selectedTeacher });
  }

  handleStartChange(newDateTime) {
    const start = moment(newDateTime).format();
    this.setState({
      start,
    });
  }

  handleEndChange(newDateTime) {
    const end = moment(newDateTime).format();
    this.setState({
      end,
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    const {
      event_id,
      start,
      end,
      type,
      rating,
      selectedLocation,
      selectedTeacher,
    } = this.state;

    const newLesson = {
      start,
      end,
      type,
      rating: Number(rating),
      location_id: selectedLocation.value,
      teacher_id: selectedTeacher.value,
    };

    editLesson(newLesson, event_id) // PUT edited lesson
      .then(this.transformServerDataIntoState);
  }

  transformServerDataIntoState(lesson) {
    const {
      allLocations,
      allTeachers,
    } = this.state;

    // repopulate state with lesson details returned from server, to ensure
    // they are synchronised
    this.setState({ ...lesson });
    // set selectedLocation from the returned lesson_id
    const newSelectedLocation = allLocations.filter(
      location => location.value === lesson.location_id,
    )[0];
    this.setState({ selectedLocation: newSelectedLocation });

    // set selectedTeacher from the returned teacher_id
    const newSelectedTeacher = allTeachers.filter(
      teacher => teacher.value === lesson.teacher_id,
    )[0];
    this.setState({ selectedTeacher: newSelectedTeacher });
  }

  render() {
    const {
      event_id,
      lesson_id,
      allLocations,
      selectedTeacher,
      selectedLocation,
      allTeachers,
      start,
      end,
      type,
      rating,
    } = this.state;

    // need to wrap start and end in moment(), or DateTime component doesn't work
    const startMoment = moment(start);
    const endMoment = moment(end);

    return (
      <div>
        <h3>View lesson</h3>
        <div>
          {JSON.stringify({
            event_id,
            lesson_id,
            selectedTeacher,
            selectedLocation,
            start,
            end,
            type,
            rating,
          })
          }
        </div>
        <form onSubmit={this.handleSubmit}>
          <Label>
            start:
            <DateTime value={startMoment} onChange={this.handleStartChange} />
          </Label>
          <Label>
            end:
            <DateTime value={endMoment} onChange={this.handleEndChange} />
          </Label>
          <Label>
            type:
            <input type="text" name="type" value={type} onChange={this.handleChange} />
          </Label>
          <Label>
            rating:
            <StarRatingComponent
              name="rating"
              value={rating}
              starCount={5}
              onStarClick={this.onStarClick}
            />
          </Label>
          <Label>
            location:
            <Select
              name="location_id"
              value={selectedLocation}
              onChange={this.handleLocationChange}
              options={allLocations}
            />
          </Label>
          <Label>
            teacher:
            <Select
              name="teacher_id"
              value={selectedTeacher}
              onChange={this.handleTeacherChange}
              options={allTeachers}
            />
          </Label>
          <input type="submit" value="Save lesson" />
        </form>
      </div>
    );
  }
}

Lesson.propTypes = {
  match: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
};

export default Lesson;
