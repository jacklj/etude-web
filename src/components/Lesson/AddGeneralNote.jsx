import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { createNote } from '../../services/api';

class AddGeneralNote extends Component {
  constructor(props) {
    super(props);
    this.state = {
      note: '',
      score: '',
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(event) {
    event.preventDefault();
    const { note, score } = this.state;
    const { eventId } = this.props;
    const type = 'NOTE_TYPES.TECHNICAL'; // TODO 8/8/2018 all general notes technical atm
    createNote({
      note, score, type, event_id: eventId,
    }).then(res => alert(`Note submitted! ${res}`));
    console.log(event);
  }

  handleChange(event) {
    const { target } = event;
    const { name, value } = target;

    this.setState({
      [name]: value,
    });
  }

  render() {
    const { note, score } = this.state;
    return (
      <form onSubmit={this.handleSubmit}>
        <textarea name="note" value={note} onChange={this.handleChange} />
        <textarea name="score" value={score} onChange={this.handleChange} />
        <input type="submit" value="Add note" />
      </form>
    );
  }
}

AddGeneralNote.propTypes = {
  eventId: undefined,
};

AddGeneralNote.propTypes = {
  eventId: PropTypes.number,
};

export default AddGeneralNote;
