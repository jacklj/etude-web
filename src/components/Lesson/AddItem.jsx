import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Select from 'react-select';

import { fetchAllRepertoireRequest, fetchAllExercisesRequest } from '../../redux/items/items.actions';
import { selectRepertoireForDropdown, selectExercisesForDropdown } from '../../redux/items/items.selectors';

const VIEW = {
  DEFAULT: 'VIEW.DEFAULT',
  ADD_EXERCISE: 'VIEW.ADD_EXERCISE',
  ADD_PIECE: 'VIEW.ADD_PIECE',
};

const initialState = {
  exercise: null,
  piece: null,
  view: VIEW.DEFAULT,
};

class AddItem extends Component {
  constructor(props) {
    super(props);
    this.state = { ...initialState };

    this.showAddPiece = this.showAddPiece.bind(this);
    this.showAddExercise = this.showAddExercise.bind(this);
    this.cancelAddingItem = this.cancelAddingItem.bind(this);
    this.handleCustomComponentChange = this.handleCustomComponentChange.bind(this);
  }

  componentDidMount() {
    this.props.fetchAllRepertoireRequest();
    this.props.fetchAllExercisesRequest();
  }

  handleCustomComponentChange(name) {
    return value => this.setState({
      [name]: value,
    });
  }

  showAddPiece() {
    this.setState({ view: VIEW.ADD_PIECE });
  }

  showAddExercise() {
    this.setState({ view: VIEW.ADD_EXERCISE });
  }

  cancelAddingItem() {
    this.setState({ ...initialState });
  }

  render() {
    const { repertoire, exercises } = this.props;
    const { exercise, piece, view } = this.state;
    let jsx;

    switch (view) {
      case VIEW.DEFAULT:
        jsx = (
          <div>
            <button type="button" onClick={this.showAddPiece}>Add piece</button>
            <button type="button" onClick={this.showAddExercise}>Add exercise</button>
          </div>
        );
        break;
      case VIEW.ADD_PIECE:
        jsx = (
          <form>
            <Select value={piece} options={repertoire} onChange={this.handleCustomComponentChange('piece')} />
            <button type="button" onClick={this.cancelAddingItem}>Cancel</button>
            <button type="button" onClick={this.addPiece}>
              Add Piece
            </button>
          </form>
        );
        break;
      case VIEW.ADD_EXERCISE:
        jsx = (
          <form>
            <Select value={exercise} options={exercises} onChange={this.handleCustomComponentChange('exercise')} />
            <button type="button" onClick={this.cancelAddingItem}>Cancel</button>
            <button type="button" onClick={this.addExercise}>
              Add Piece
            </button>
          </form>
        );
        break;
      default:
        break;
    }
    return jsx;
  }
}

AddItem.defaultProps = {
  eventId: undefined,
};

AddItem.propTypes = {
  eventId: PropTypes.number,
  repertoire: PropTypes.array.isRequired, // eslint-disable-line react/forbid-prop-types
  exercises: PropTypes.array.isRequired, // eslint-disable-line react/forbid-prop-types
  fetchAllRepertoireRequest: PropTypes.func.isRequired,
  fetchAllExercisesRequest: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  repertoire: selectRepertoireForDropdown(state),
  exercises: selectExercisesForDropdown(state),
});

const mapDispatchToProps = {
  fetchAllRepertoireRequest,
  fetchAllExercisesRequest,
};

export default connect(mapStateToProps, mapDispatchToProps)(AddItem);
