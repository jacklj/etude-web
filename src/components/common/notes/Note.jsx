import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { Card, Label } from '../styledComponents';

import { updateNoteRequest, deleteNoteRequest } from '../../../redux/notes/notes.actions';

const VIEW = {
  DISPLAY: 'VIEW.DISPLAY',
  EDIT: 'VIEW.EDIT',
  DELETE: 'VIEW.DELETE',
};

export class Note extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentView: VIEW.DISPLAY,
      editingNote: '',
      editingScore: '',
    };

    this.showDeleteNote = this.showDeleteNote.bind(this);
    this.showEditNote = this.showEditNote.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.deleteNote = this.deleteNote.bind(this);
    this.editNoteSubmit = this.editNoteSubmit.bind(this);
    this.cancelEditingOrDeletingNote = this.cancelEditingOrDeletingNote.bind(this);
  }

  showDeleteNote() {
    this.setState({
      currentView: VIEW.DELETE,
    });
  }

  deleteNote(event) {
    event.preventDefault();
    const { noteId } = this.props;
    this.props.deleteNoteRequest(noteId);
    // no need to change the view, as this component will disappear!
  }

  showEditNote() {
    const { note, score } = this.props;
    this.setState({
      currentView: VIEW.EDIT,
      editingNote: note,
      editingScore: score,
    });
  }

  handleChange(event) {
    const { target } = event;
    const { name, value } = target;

    this.setState({
      [name]: value,
    });
  }

  editNoteSubmit(event) {
    event.preventDefault();
    const {
      editingNote,
      editingScore,
    } = this.state;

    const updatedNote = {
      note: editingNote,
      score: editingScore,
      type: this.props.type,
      note_id: this.props.noteId,
    };

    this.props.updateNoteRequest(updatedNote, this.props.noteId);
    this.setState({ currentView: VIEW.DISPLAY });
  }

  cancelEditingOrDeletingNote() {
    this.setState({ currentView: VIEW.DISPLAY });
  }

  render() {
    const { note, score, type } = this.props;
    const { currentView, editingNote, editingScore } = this.state;

    let jsx;
    switch (currentView) {
      case VIEW.DISPLAY:
        jsx = (
          <div>
            <div>{note}</div>
            <div>{score}</div>
            <div>{type}</div>
            {this.props.editable && (
              <div>
                <button type="button" onClick={this.showEditNote}>Edit</button>
                <button type="button" onClick={this.showDeleteNote}>Delete</button>
              </div>
            )}
          </div>
        );
        break;
      case VIEW.EDIT:
        jsx = (
          <form onSubmit={this.editNoteSubmit}>
            <div>editing</div>
            <Label>
              note:
              <textarea name="editingNote" value={editingNote} onChange={this.handleChange} />
            </Label>
            <Label>
              score:
              <textarea name="editingScore" value={editingScore} onChange={this.handleChange} />
            </Label>
            <input type="submit" value="Save" />
            <button type="button" onClick={this.cancelEditingOrDeletingNote}>Cancel</button>
          </form>
        );
        break;
      case VIEW.DELETE:
        jsx = (
          <div>
            <button type="button" onClick={this.cancelEditingOrDeletingNote}>Cancel</button>
            <button type="button" onClick={this.deleteNote}>Delete</button>
          </div>
        );
        break;
      default:
        break;
    }
    return (
      <Card>
        {jsx}
      </Card>
    );
  }
}

Note.defaultProps = {
  score: '',
  editable: true,
};

Note.propTypes = {
  note: PropTypes.string.isRequired,
  score: PropTypes.string,
  type: PropTypes.string.isRequired,
  noteId: PropTypes.number.isRequired,
  updateNoteRequest: PropTypes.func.isRequired,
  deleteNoteRequest: PropTypes.func.isRequired,
  editable: PropTypes.bool,
};

const mapStateToProps = () => ({});

const mapDispatchToProps = {
  updateNoteRequest,
  deleteNoteRequest,
};

export default connect(mapStateToProps, mapDispatchToProps)(Note);
