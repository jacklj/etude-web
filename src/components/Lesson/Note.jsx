import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { Card } from '../common/itemCards';
import { Label } from '../common/itemSections';

import { generalNoteUpdateRequest } from '../../redux/notes/notes.actions';

export class Note extends Component {
  constructor(props) {
    super(props);
    this.state = {
      editing: false,
      editingNote: '',
      editingScore: '',
    };

    this.editNote = this.editNote.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.cancelEditingNote = this.cancelEditingNote.bind(this);
  }

  editNote() {
    const { note, score } = this.props;
    this.setState({
      editing: true,
      editingNote: note,
      editingScore: score,
    });
  }

  handleChange(event) {
    const { target } = event;
    const { name, value } = target;

    this.setState({
      [name]: value,
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    const {
      editingNote,
      editingScore,
    } = this.state;
    const { id } = this.props;

    const updatedNote = {
      note: editingNote,
      score: editingScore,
      type: this.props.type,
      id: this.props.id,
      event_id: this.props.eventId,
    };

    this.props.generalNoteUpdateRequest(updatedNote, id);
    this.setState({ editing: false });
  }

  cancelEditingNote() {
    this.setState({ editing: false });
  }

  render() {
    const { note, score, type } = this.props;
    const { editing, editingNote, editingScore } = this.state;
    return (
      <Card>
        {editing ? (
          <form onSubmit={this.handleSubmit}>
            <div>editing</div>
            <Label>
              note:
              <textarea name="editingNote" value={editingNote} onChange={this.handleChange} />
            </Label>
            <Label>
              score:
              <textarea name="editingScore" value={editingScore} onChange={this.handleChange} />
            </Label>
            <input type="submit" value="Save" />
            <button type="button" onClick={this.cancelEditingNote}>Cancel</button>
          </form>
        ) : (
          <div>
            <div>{note}</div>
            <div>{score}</div>
            <div>{type}</div>
            <button type="button" onClick={this.editNote}>Edit</button>
          </div>
        )
        }
      </Card>
    );
  }
}

Note.defaultProps = {
  score: '',
};

Note.propTypes = {
  note: PropTypes.string.isRequired,
  score: PropTypes.string,
  type: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  eventId: PropTypes.number.isRequired,
  generalNoteUpdateRequest: PropTypes.func.isRequired,
};

const mapStateToProps = () => ({});

const mapDispatchToProps = {
  generalNoteUpdateRequest,
};

export default connect(mapStateToProps, mapDispatchToProps)(Note);
