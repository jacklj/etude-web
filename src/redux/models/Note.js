import { fk, attr, Model } from 'redux-orm';

import { ACTION_TYPES } from '../notes/notes.actions';

class Note extends Model {
  static reducer(action, Note, session) {
    switch (action.type) {
      case ACTION_TYPES.NOTE.CREATE_AND_ADD_TO.EVENT.SUCCESS:
        Note.create(action.note);
        break;
      default:
        break;
    }
    // Return value is ignored.
    return undefined;
  }

  toString() {
    return `Note: ${this.note}`;
  }
  // Declare any static or instance methods you need.
}
Note.modelName = 'Note';

// Declare your related fields.
Note.fields = {
  note_id: attr(), // non-relational field for any value; optional but highly recommended
  note: attr(),
  score: attr(),
  type: attr(),
  event_id: fk({ to: 'Event', relatedName: 'notes' }),
};

Note.options = {
  idAttribute: 'note_id',
};

export default Note;
