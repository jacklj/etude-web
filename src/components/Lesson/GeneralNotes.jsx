import React from 'react';
import PropTypes from 'prop-types';

import Note from './Note';

const GeneralNotes = ({ notes }) => (
  <div>
    <h3>Notes</h3>
    {notes
      && Object.values(notes).map(note => (
        <Note
          key={note.note_id}
          note={note.note}
          score={note.score}
          type={note.type}
          id={note.note_id}
          eventId={note.event_id}
        />
      ))}
  </div>
);

GeneralNotes.defaultProps = {
  notes: undefined,
};

GeneralNotes.propTypes = {
  notes: PropTypes.object, // eslint-disable-line react/forbid-prop-types
};

export default GeneralNotes;
