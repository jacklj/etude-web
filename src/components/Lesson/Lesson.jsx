/* eslint-disable camelcase */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import moment from 'moment';
import StarRatingComponent from 'react-star-rating-component';
import DateTime from 'react-datetime';
import Select from 'react-select';

import Items from '../Items';
import AddGeneralNote from './AddGeneralNote';
import GeneralNotes from './GeneralNotes';
import { Label } from '../common/itemSections';
import { locationsFetchRequest } from '../../redux/locations/locations.actions';
import { peopleFetchRequest } from '../../redux/people/people.actions';
import { selectLocationsForDropdown } from '../../redux/locations/locations.selectors';
import { selectTeachersForDropdown } from '../../redux/people/people.selectors';
import { selectEvent } from '../../redux/events/events.selectors';
import {
  editLesson, getLesson,
} from '../../services/api';
import { renderDuration } from '../../services/datetime';

class Lesson extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedLocation: undefined,
      selectedTeacher: undefined,
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.onStarClick = this.onStarClick.bind(this);
    this.handleStartChange = this.handleStartChange.bind(this);
    this.handleEndChange = this.handleEndChange.bind(this);
    this.handleLocationChange = this.handleLocationChange.bind(this);
    this.handleTeacherChange = this.handleTeacherChange.bind(this);
    this.transformServerDataIntoState = this.transformServerDataIntoState.bind(this);
  }

  componentDidMount() {
    const { eventId } = this.props;
    this.setState({ event_id: eventId });

    this.props.locationsFetchRequest();
    this.props.peopleFetchRequest();

    // getLesson to ensure it's up to date in the store
    this.transformServerDataIntoState(this.props.lesson);
  }

  onStarClick(nextValue) {
    this.setState({ rating: nextValue });
  }

  handleChange(event) {
    const { target } = event;
    const { name, value } = target;

    this.setState({
      [name]: value,
    });
  }

  handleLocationChange(selectedLocation) {
    this.setState({ selectedLocation });
  }

  handleTeacherChange(selectedTeacher) {
    this.setState({ selectedTeacher });
  }

  handleStartChange(newDateTime) {
    const start = moment(newDateTime).format();
    this.setState({
      start,
    });
  }

  handleEndChange(newDateTime) {
    const end = moment(newDateTime).format();
    this.setState({
      end,
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    const {
      event_id,
      start,
      end,
      type,
      rating,
      selectedLocation,
      selectedTeacher,
    } = this.state;

    const newLesson = {
      start,
      end,
      type,
      rating: Number(rating),
      location_id: selectedLocation.value,
      teacher_id: selectedTeacher.value,
    };

    editLesson(newLesson, event_id) // PUT edited lesson
      .then(this.transformServerDataIntoState);
  }

  transformServerDataIntoState(lesson) {
    // repopulate state with lesson details returned from server, to ensure
    // they are synchronised
    this.setState({ ...lesson });
    // set selectedLocation from the returned lesson_id
    const newSelectedLocation = this.props.locations.filter(
      location => location.value === lesson.location.id,
    )[0];
    this.setState({ selectedLocation: newSelectedLocation });

    // set selectedTeacher from the returned teacher_id
    const newSelectedTeacher = this.props.teachers.filter(
      teacher => teacher.value === lesson.teacher.id,
    )[0];
    this.setState({ selectedTeacher: newSelectedTeacher });
  }

  render() {
    const {
      event_id,
      selectedTeacher,
      selectedLocation,
      start,
      teacher,
      end,
      type,
      rating,
      items,
      general_notes,
    } = this.state;

    // need to wrap start and end in moment(), or DateTime component doesn't work
    const startMoment = moment(start);
    const endMoment = moment(end);

    return (
      <div>
        <h3>Lesson{teacher && ` with ${teacher.first_name} ${teacher.surname}`}, {renderDuration(start, end)}</h3>
        <div>
          {JSON.stringify(this.state)}
        </div>
        <form onSubmit={this.handleSubmit}>
          <Label>
            start:
            <DateTime value={startMoment} onChange={this.handleStartChange} />
          </Label>
          <Label>
            end:
            <DateTime value={endMoment} onChange={this.handleEndChange} />
          </Label>
          <Label>
            type:
            <input type="text" name="type" value={type} onChange={this.handleChange} />
          </Label>
          <Label>
            rating:
            <StarRatingComponent
              name="rating"
              value={rating}
              starCount={5}
              onStarClick={this.onStarClick}
            />
          </Label>
          <Label>
            location:
            <Select
              name="location_id"
              value={selectedLocation}
              onChange={this.handleLocationChange}
              options={this.props.locations}
            />
          </Label>
          <Label>
            teacher:
            <Select
              name="teacher_id"
              value={selectedTeacher}
              onChange={this.handleTeacherChange}
              options={this.props.teachers}
            />
          </Label>
          <input type="submit" value="Save details" />
        </form>
        <h3>General lesson notes</h3>
        <AddGeneralNote eventId={event_id} />
        <GeneralNotes notes={general_notes} />
        <Items items={items} />
      </div>
    );
  }
}

Lesson.propTypes = {
  eventId: PropTypes.number.isRequired,
  lesson: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
  locations: PropTypes.array.isRequired, // eslint-disable-line react/forbid-prop-types
  locationsFetchRequest: PropTypes.func.isRequired,
  peopleFetchRequest: PropTypes.func.isRequired,
  teachers: PropTypes.array.isRequired, // eslint-disable-line react/forbid-prop-types
};

const mapStateToProps = (state, ownProps) => ({
  // ownProps isn't recursive - just props supplied from 'above'
  eventId: ownProps.match.params.id,
  locations: selectLocationsForDropdown(state),
  teachers: selectTeachersForDropdown(state),
  lesson: selectEvent(state, ownProps),
});

const mapDispatchToProps = {
  locationsFetchRequest,
  peopleFetchRequest,
};

export default connect(mapStateToProps, mapDispatchToProps)(Lesson);
