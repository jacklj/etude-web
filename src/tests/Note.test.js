/* eslint-disable react/jsx-filename-extension */
import renderer from 'react-test-renderer';

import React from 'react';
import { Note } from '../components/Lesson/Note';

test('Note renders correctly', () => {
  const component = renderer.create(
    <Note
      key={1}
      note="A note"
      score="a b c"
      type="TECHNICAL"
      id={1}
      eventId={1}
      generalNoteUpdateRequest={() => ({})}
      noteDeleteRequest={() => ({})}
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // // manually trigger the callback
  // tree.props.onMouseEnter();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
  //
  // // manually trigger the callback
  // tree.props.onMouseLeave();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});
