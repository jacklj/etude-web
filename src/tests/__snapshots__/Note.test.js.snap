// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note renders correctly 1`] = `
<div
  className="sc-bdVaJa jqsMAW"
>
  <div>
    <div>
      A note
    </div>
    <div>
      a b c
    </div>
    <div>
      TECHNICAL
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      Edit
    </button>
  </div>
</div>
`;
